packages/core/src/config/config.ts

```diff
@@ -0,0 +1,161 @@
+import { ConfigUrl } from '../index.js';
+import type {
+  Domain,
+  Domainlike,
+  EthereumConfig,
+  Network,
+  Resource,
+  SubstrateConfig,
+  SygmaConfig,
+} from '../types.js';
+import { Environment } from '../types.js';
+
+import { localConfig } from './localConfig.js';
+
+export class Config {
+  /**
+   * Cache storing all sygma
+   * environment configurations
+   */
+  configuration!: SygmaConfig;
+  initialized = false;
+
+  constructor() {}
+  /**
+   * Initialize and store all
+   * sygma configurations
+   */
+  async init(environment: Environment): Promise<void> {
+    if (environment === Environment.LOCAL) {
+      this.configuration = localConfig;
+    } else {
+      const response = await fetch(this.getConfigUrl(environment));
+      const data = (await response.json()) as SygmaConfig;
+      this.configuration = data;
+      // initialized is set to true when
+      // all configurations have been fetched
+      this.initialized = true;
+    }
+  }
+  /**
+   * Retrieve hosted bridge configuration in JSON format
+   * @param {Environment} environment bridging environment i.e mainnet, testnet or devnet
+   * @returns {Promise<SygmaConfig>}
+   */
+  private getConfigUrl(environment: Environment): string {
+    switch (environment) {
+      case Environment.DEVNET: {
+        return ConfigUrl.DEVNET;
+      }
+      case Environment.TESTNET: {
+        return ConfigUrl.TESTNET;
+      }
+      default:
+        return ConfigUrl.MAINNET;
+    }
+  }
+  /**
+   * Creates a domain object from config object
+   * @param {EthereumConfig | SubstrateConfig} config
+   * @returns {Domain}
+   */
+  private createDomain(config: EthereumConfig | SubstrateConfig): Domain {
+    return {
+      id: config.id,
+      caipId: config.caipId,
+      chainId: config.chainId,
+      name: config.name,
+      type: config.type,
+      parachainId: (config as SubstrateConfig).parachainId,
+    };
+  }
+  findDomainConfigBySygmaId(sygmaId: number): SubstrateConfig | EthereumConfig {
```

Comment:
there are cases in fee processing when we have to explicitly search on the basis of sygma ID